src/services/sync/strategies/consumerActions/ClassAction.js

```diff
@@ -0,0 +1,48 @@
+const BaseConsumerAction = require('./BaseConsumerAction');
+// TODO: place from where it is importat must be fixed later
+const { LDAP_SYNC_ACTIONS } = require('../LDAPSyncer');
+const { SchoolRepo, ClassRepo } = require('../../repo');
+
+const defaultOptions = {
+	allowedLogKeys: ['class', 'ldapDN', 'systemId', 'schoolDn', 'year'],
```

Comment:
here were repos before. Why repository definitions are necessary here? We can use Repositories directly without 1this1.

Reply:
To make it possible to replace repos without sinon or other stepping tools for test. To replace it for other use cases if needed. Datalayer that fetch data from other db for example. Is more like make it much easyer as a really need and bring some complexity in the BaseConsumerAction class.

I put it into it, because i really like if we can put a lot of easy readable error test ...and for my personal view this means sinon is not a option.

Reply:
But in the end we must commit us of what way we want to go. Dependency injection for repos possible yes or no.

Reply:
This is completelty new concept. We should discuss it. I don't think that it should be part of this PR.